build-all.sh

```diff
@@ -599,6 +612,13 @@ then
     fi
 fi
 
+if [ "x${tls_support}" = "x--with-tls" ]
+then
+    TLS_SUPPORT="yes"
+else
+    TLS_SUPPORT="no"
+fi
+
```

Comment:
Perhaps, you can have `TLS_SUPPORT="no"` as a default, and then move this `if` into the option parsing `case`, to make things simplier. Or even better:
```
--tls) TLS_SUPPORT=yes ;;
--no-tls) TLS_SUPPORT=no ;;
```
That looks cleaner to me. Most of option handing after the case is there because of cross-option dependies, so we cannot do anything until all options has been parsed (there are some exception, which can be moved to `case` as well).